composer/trainer/dist_strategy.py

```diff
@@ -459,7 +461,28 @@ def __auto_wrap_policy(module: torch.nn.Module, recurse: bool, nonwrapped_numel:
                 def _auto_wrap_policy_new(module: torch.nn.Module, recurse: bool, nonwrapped_numel: int) -> bool:
                     return __auto_wrap_policy(module, recurse, nonwrapped_numel)
 
-                _auto_wrap_policy = _auto_wrap_policy_new
+                if version.parse(torch.__version__) > version.parse('2.0.1-rc4'):
```

Comment:
Should this be 2.1.0?

Reply:
`version` compares somewhat lexicographically, e.g. version.parse('2.1.0-rc1') <  version.parse('2.1.0') or version.parse('2.1.0-dev') <  version.parse('2.1.0')

Reply:
@cli99  I think this is resulting in this code path being triggered even for `2.0.1`. It is currently `2.0.1-rc4` where I think it should be `2.1.0-rc4`

Reply:
@mvpatel2000, you are right. However, for torch.__version_ == '2.1.0.devXXX', the condition is still evaluated to false if we use `2.1.0-rc4`, maybe we shall use `if version.parse(torch.__version__) > version.parse('2.1.0.dev')`:

Reply:
That seems reasonable to me! if it works let's use it